### PR TITLE
[integration] py3 only fixes

### PR DESCRIPTION
**Pull request by fstagni**
This PR contains several code fixes for python 3 only. They don't deserve release notes.

### CODE REVIEW

File: src/DIRAC/Core/Utilities/DIRACSingleton.py [hunk 1]
```diff
@@ -36,11 +24,9 @@ class DIRACSingleton(type):
     If you want make your class a singleton, just set its  __metaclass__ to
```
Comment (andresailer) [inline src/DIRAC/Core/Utilities/DIRACSingleton.py:24]:
```suggestion
    If you want make your class a singleton, just set the keyword argument ``metaclass=DIRACSingleton``  in its definition
```
next line also needs update.

File: src/DIRAC/FrameworkSystem/Client/SystemAdministratorClientCLI.py [hunk 2]
```diff
@@ -1068,23 +1065,20 @@ def do_stop(self, args):
...
-              where lcgVersion - version of DIRACOS to install
+          update <version>
         """
         try:
             argss = args.split()
             version = argss[0]
-            diracOSVersion = ""
             del argss[0]
 
             while len(argss) > 0:
```
Comment (andresailer) [inline src/DIRAC/FrameworkSystem/Client/SystemAdministratorClientCLI.py:1080]: this whole `while` can be removed

File: src/DIRAC/Core/Utilities/DIRACSingleton.py [hunk 3]
```diff
@@ -36,11 +24,9 @@ class DIRACSingleton(type):
...
     DIRACSingleton, i.e.::
 
-      import six
       from DIRAC.Core.Utilities.DIRACSingleton import DIRACSingleton
-      @six.add_metaclass(DIRACSingleton)
-      class CheesShop(object):
-        ...
+      class CheesShop(metaclass=DIRACSingleton):
```
Comment (andresailer) [inline src/DIRAC/Core/Utilities/DIRACSingleton.py:28]:
```suggestion
      class CheeseShop(metaclass=DIRACSingleton):
```

File: src/DIRAC/Core/Base/SQLAlchemyDB.py [hunk 4]
```diff
@@ -9,7 +9,6 @@
 
 __RCSID__ = "$Id$"
```
Comment (atsareg) [inline src/DIRAC/Core/Base/SQLAlchemyDB.py:10]: I guess all the occurrences of __RCSID__ can be deleted now as they are no filled in by the release procedure

File: src/DIRAC/Core/Utilities/Graph.py [hunk 5]
```diff
@@ -74,405 +68,4 @@ def _getProperty(self, name):
         classdict["makeProperty"] = makeProperty
         classdict["_setProperty"] = _setProperty
         classdict["_getProperty"] = _getProperty
-        return type.__new__(cls, name, bases, classdict)
-
-
-@six.add_metaclass(DynamicProps)
```
Comment (atsareg) [inline src/DIRAC/Core/Utilities/Graph.py:80]: The only left class here is DynamicProps which in turn is only used in OperationHandlerBase.py. It is better to move this class to the OperationHandlerBase.py and scrap Utilities/Graph.py altogether

File: src/DIRAC/Core/Utilities/test/Test_Graph.py [hunk 6]
```diff
@@ -1,32 +1,14 @@
...
-# @author Krzysztof.Ciba@NOSPAMgmail.com
-# @date 2012/09/28 09:02:24
-# @brief Definition of GraphTests class.
-# # imports
 import unittest
-import six
 
 # # SUT
-from DIRAC.Core.Utilities.Graph import Node, Edge, Graph, DynamicProps  # , topologicalSort, topoSort
+from DIRAC.Core.Utilities.Graph import DynamicProps
```
Comment (atsareg) [inline src/DIRAC/Core/Utilities/test/Test_Graph.py:11]: If DIRAC.Core.Utilities.Graph is removed, this test can be also removed

**Comment by atsareg** [comment]
Impressive. The PR looks fine to me. Please just consider a couple of inline comments.

File: src/DIRAC/Core/Utilities/Graphs/GraphUtilities.py [hunk 7]
```diff
@@ -75,13 +67,19 @@ def convert_to_datetime(dstring):
...
         else:
             results = eval(str(dstring), {"__builtins__": None, "time": time, "math": math}, {})
-        if isinstance(results, six.integer_types + (float,)):
+        if isinstance(
+            results,
+            (
+                int,
+                float,
+            ),
+        ):
```
Comment (chrisburr) [inline src/DIRAC/Core/Utilities/Graphs/GraphUtilities.py:76]: The trailing comma is significant and cause black to wrap the lines, I think this is clearer:
```suggestion
        if isinstance(results, (int, float)):
```

File: src/DIRAC/FrameworkSystem/Service/SystemAdministratorHandler.py [hunk 8]
```diff
@@ -301,17 +281,13 @@ def export_updateSoftware(self, version, _rootPath="", diracOSVersion=""):
...
 
         # Install DIRACOS
-        installer_url = "https://github.com/DIRACGrid/DIRACOS2/releases/"
-        if diracOSVersion:
-            installer_url += "download/%s/latest/download/DIRACOS-Linux-%s.sh" % (version, platform.machine())
-        else:
-            installer_url += "latest/download/DIRACOS-Linux-%s.sh" % platform.machine()
+        installer_url = (
+            "https://github.com/DIRACGrid/DIRACOS2/releases/latest/download/DIRACOS-Linux-%s.sh" % platform.machine()
+        )
```
Comment (chrisburr) [inline src/DIRAC/FrameworkSystem/Service/SystemAdministratorHandler.py:290]: Do we want to remove this functionality for setting the version?

**Review by chrisburr — APPROVED** [review]
Looks good to me, I have to minor comments but I'd be in favour of merging so we can test with it tomorrow.

**Comment by fstagni** [comment]
I merge now, and I will take care of the comments in the next PR.

**Pull request merged**